symphonia-core/src/units.rs

```diff
@@ -70,6 +70,10 @@ impl Time {
 
         Some(Time { seconds, frac })
     }
+
+    pub fn into_duration(&self) -> StdDuration {
+        StdDuration::from_secs_f64(self.seconds as f64 + self.frac)
```

Comment:
A `f64` can only represent an unsigned integer up-to 2^52 without losing precision, so in extreme cases where seconds is very large, this conversion will be lossy.

Using `Duration::new` and passing the `u64` seconds component directly will preserve the accuracy in those extreme cases. To implement this, the fractional part must be converted to nanoseconds.

```suggestion
        StdDuration::new(self.seconds, (1_000_000_000.0 * self.frac) as u32)
```

Reply:
Done.